### ISSUE TITLE
Crate a bridge with translation services

### ISSUE THREAD

**Issue by javigomez** [joomla-projects/com_localise#254]
This is an experimental feature but it would be awesome that com_localise will be able to have an API that allows it to send and retrieve translations from services like Transifex or Crowdin.

Adding here so we don't forget what we spoke during JWC14 @yireo

**Comment by infograf768** [comment]
Transifex is not good enough for J core as it does not take care of supplementary strings depending on languages (plurals, others)

**Comment by javigomez** [comment]
Yes, Transifex can't do this: https://github.com/joomla-projects/translate-joomla#working-with-plurals

Still having an API will allow us to export and import files and do crazy things. Having a webservices layer for com_localise would allow us to simplify tasks like https://github.com/joomla-projects/com_localise/issues/57 or for example to connect it with Slack, so any new released language package sends a notification to Slack.

This feature is not necessarily a bridge with transifex but an experimentation with opening the tool for interacting with other tools.

**Comment by fastslack** [comment]
This can be done using some ajax.raw.php and returning the required data via JSON. What kind of data we need to export?

**Comment by javigomez** [comment]
Well, is not clear yet. But actually I can think on one translation. That would allow us to do inline edition of translations, or complete files.

But you are right we will need to think it what thinks to we want to allow to be in the api: translation, language file, package...

**Comment by elkuku** [comment]
> Transifex is not good enough for J core as it does not take care of supplementary strings depending on languages (plurals, others)

With "others" you probably mean the special `QQ` constant?
As this is about to disappear (hopefully), Transifex maybe now "good enough" for our needs?
Not sure that Transifex really supports the special plural string handling but hey...

@javigomez this is an awesome idea :wink:

**Comment by infograf768** [comment]
> With "others" you probably mean the special QQ constant?

Nope. I mean specific language strings as (in fr-FR):

```
MOD_MENU_HELP_SUPPORT_CUSTOM_FORUM="Forum joomla.fr"
MOD_MENU_HELP_SUPPORT_OFFICIAL_LANGUAGE_FORUM="Forum officiel français"
MOD_MENU_HELP_SUPPORT_OFFICIAL_LANGUAGE_FORUM_VALUE="19"
```

or

```
INSTL_SAMPLE_BLOG_FR_SET="Données exemples de type blog, en français"
INSTL_SAMPLE_BROCHURE_FR_SET="Données exemples de type brochure, en français"
INSTL_SAMPLE_DATA_FR_SET="Simple page d'accueil, en français"
INSTL_SAMPLE_LEARN_FR_SET="Données exemples pour apprendre Joomla, en français"
INSTL_SAMPLE_TESTING_FR_SET="Données exemples pour tester Joomla, en français"
INSTL_SAMPLE_BLOG_FR_SET_DESC="Installer Joomla avec quelques articles et des modules de type blog tels que Articles Archivés, Articles les plus consultés, etc."
INSTL_SAMPLE_BROCHURE_FR_SET_DESC="Installer Joomla avec quelques pages (un menu avec Page d'acceuil, A propos de nous, Nouvelles, Nous contacter) et des modules comme Recherche, Contenu personnalisé, Formulaire de connexion."
INSTL_SAMPLE_DATA_FR_SET_DESC="Installer Joomla avec une page (un menu avec un lien) et des modules comme Derniers Articles, Formulaire de connexion."
INSTL_SAMPLE_LEARN_FR_SET_DESC="Installer Joomla avec des exemples d'articles décrivant la façon dont Joomla fonctionne."
INSTL_SAMPLE_TESTING_FR_SET_DESC="Installer Joomla avec tous les liens de menu possible pour aider à tester Joomla."
```

and others in core... for example reCaptcha version1 where the strings are commented for some languages but should not for others, etc.
